### PR TITLE
Task_9993_9998_10003: Edit Genres, Settings, Term, BookShelfCategories Screen

### PR DESCRIPTION
**Pull request by khanhnn-1791**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9993
https://edu-redmine.sun-asterisk.vn/issues/9998
https://edu-redmine.sun-asterisk.vn/issues/10003
### Goals :soccer:

https://marvelapp.com/project/3891276/canvas/55157798/

Create UI for Genres Screen
+ Close, Save label
+ Navigation bar title
+ Genres list using collectionview

Create a UI action
- Click close to go to the previous view 
- Click Done to save and go to the previous view
- Select genre cell and change color
Fetch data from API and display to the UI

https://marvelapp.com/project/3891276/screen/55143240
Create UI for Term of User Screen
+ Close Label
+ Navigation bar title
+ Content Label
Create a UI action
- Click Close to go to the previous view 

https://marvelapp.com/project/3891276/screen/55070066
Create UI for Setting Screen
+ Back Label
+ Navigation bar title
+ Term of use label/ icon
Create a UI action
- Click Back to go to the previous view 
- Click Term of use to go to the term of use screen
Fetch data from API and display to the UI
### Implementation Details :construction:

https://gfycat.com/SizzlingSimplisticAiredaleterrier

### CODE REVIEW

**Comment by khanhnn-1791** [comment]
ready

File: ios-i-book-lover/Scenes/Profile/BookShelfCategoryViewController.swift [hunk 1]
```diff
@@ -8,11 +8,95 @@
...
+}
+
+extension BookShelfCategoryViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return bookList.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(for: indexPath) as BookListCell
+        if !bookList.isEmpty {
```
Comment (dongsonbk58) [inline ios-i-book-lover/Scenes/Profile/BookShelfCategoryViewController.swift]: sao phải check cái này..nếu empty thì nó không vô đây đâu.
bỏ đi em nhé

File: ios-i-book-lover/Scenes/Cell/GenreCell/GenreEditCell.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import UIKit
+import Reusable
+
+final class GenreEditCell: UICollectionViewCell, Reusable {
+    @IBOutlet private weak var genreLabel: UILabel!
+    
+    override var isSelected: Bool {
+        didSet {
+            if isSelected {
```
Comment (longvt96) [inline ios-i-book-lover/Scenes/Cell/GenreCell/GenreEditCell.swift]:
```suggestion 
genreLabel.do {
         $0.backgroundColor = isSelected ? .mainColor :  .white
         $0.textColor = isSelected ? .white : .mainColor
}
```

File: ios-i-book-lover/Scenes/Profile/EditGernesViewController.swift [hunk 3]
```diff
@@ -8,11 +8,121 @@
...
+
+extension EditGernesViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return genres.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(for: indexPath) as GenreEditCell
+        let genre = genres[indexPath.item]
+        if !selectedIndexGenres.isEmpty {
```
Comment (dongsonbk58) [inline ios-i-book-lover/Scenes/Profile/EditGernesViewController.swift]: tương tự như trên

File: ios-i-book-lover/Scenes/Cell/GenreCell/GenreEditCell.swift [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        super.awakeFromNib()
+        configCell()
+    }
+    
+    func setContent(genreTitle: String) {
+        genreLabel.text = genreTitle
+    }
+    
+    private func configCell() {
+        genreLabel.do { _ in
```
Comment (longvt96) [inline ios-i-book-lover/Scenes/Cell/GenreCell/GenreEditCell.swift]: chỗ này bên trong do k dùng đến label thì cần gì viết genreLabel.do  nhỉ?

**Review by longvt96 — APPROVED** [review]

**Review by dongsonbk58 — APPROVED** [review]

File: ios-i-book-lover/Scenes/Profile/BookShelfCategoryViewController.swift [hunk 5]
```diff
@@ -8,11 +8,93 @@
...
+    var categoryProgress = -1
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        prepareUI()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        fetchData()
```
Comment (tienpx3010) [inline ios-i-book-lover/Scenes/Profile/BookShelfCategoryViewController.swift:27]: viewWillAppear

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**